### PR TITLE
[Feature] add support for partition_ttl  on materialized view

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

Introduce the `partition_ttl` property for Materialized View:
```sql
CREATE MATERIALIZED VIEW test.mv_ttl_mv1
 REFRESH ASYNC 
 PARTITION BY k1
 PROPERTIES('partition_ttl'='2 month')
 AS SELECT k1, v1 FROM test.tbl1
```

The `partition_ttl='2 month'` means, retain partitions whose data is within two months.

At the meantime, the `partition_ttl_number` is not recommended.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 59 / 67 (88.06%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|6|9|66.67%|[328, 329, 334]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|8|87.50%|[3100]|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|28|32|87.50%|[531, 542, 546, 555]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DynamicPartitionUtil.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterMVJobExecutor.java|10|10|100.00%|[]|

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/clone/DynamicPartitionScheduler.java [hunk 1]
```diff
@@ -513,6 +520,51 @@ private void executePartitionTimeToLive() {
...
+     * Drop the partition if partition upper endpoint less than TTL lower bound
+     */
+    private ArrayList<DropPartitionClause> buildDropPartitionClauseByTTL(OlapTable olapTable,
+                                                                         PeriodDuration ttlDuration)
+            throws AnalysisException {
+        if (ttlDuration.isZero()) {
+            return Lists.newArrayList();
+        }
+        ArrayList<DropPartitionClause> dropPartitionClauses = new ArrayList<>();
+        RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) (olapTable.getPartitionInfo());
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/clone/DynamicPartitionScheduler.java:534]: It may not be `RangePartitionInfo` ?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/clone/DynamicPartitionScheduler.java:534]: only range partition is supported

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 73 / 83 (87.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|6|9|66.67%|[333, 334, 339]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|8|87.50%|[3209]|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|28|32|87.50%|[531, 542, 546, 555]|
|:large_blue_circle:|com/starrocks/sql/common/PartitionDiffer.java|14|16|87.50%|[126, 130]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DynamicPartitionUtil.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterMVJobExecutor.java|10|10|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.2 branch-3.1

**Pull request merged**